adapter/pkg/messaging/notificationListener.go

```diff
@@ -90,71 +100,69 @@ func handleNotification(deliveries <-chan amqp.Delivery, done chan error) {
 	}
 
-	logger.LoggerMsg.Infof("oldTimeStamp: %v , newTimeStamp: %v", oldTimeStamp, newTimeStamp)
-	if isFound && oldTimeStamp < newTimeStamp && strings.EqualFold(removeAPIFromGateway, apiEvent.Event.Type) {
-		deleteAPIFromList(indexOfAPI, apiEvent.APIID)
-	} else if strings.EqualFold(deployAPIToGateway, apiEvent.Event.Type) {
-		// pull API details
-		ID, err := strconv.ParseInt(apiEvent.APIID, 01, 32)
-		if err != nil {
-			logger.LoggerMsg.Errorf("Cannot cast %s to an Integer", apiEvent.APIID)
-		} else {
-			api := resourceTypes.API{APIID: strconv.FormatInt(ID, 10), UUID: apiEvent.UUID,
-				Provider: apiEvent.APIProvider, Name: apiEvent.APIName,
-				Version: apiEvent.APIVersion, Context: apiEvent.APIContext, APIType: apiEvent.APIType,
-				APIStatus: apiEvent.APIStatus, IsDefaultVersion: true, TenantID: apiEvent.TenantID,
-				TenantDomain: apiEvent.Event.TenantDomain, TimeStamp: apiEvent.Event.TimeStamp}
-
-			if apiEvent.Event.Type == "API_CREATE" {
-				subscription.APIList.List = append(subscription.APIList.List, api)
-			} else if apiEvent.Event.Type == "API_UPDATE" {
-				subscription.APIList.List = removeAPI(subscription.APIList.List, apiEvent.APIID)
-				subscription.APIList.List = append(subscription.APIList.List, api)
-			} else if apiEvent.Event.Type == "API_DELETE" {
-				subscription.APIList.List = removeAPI(subscription.APIList.List, apiEvent.APIID)
+	// TODO: (VirajSalaka) Handle API Blocked event
+	if len(apiEvent.GatewayLabels) > 0 {
+		for _, env := range apiEvent.GatewayLabels {
+			// TODO: (VirajSalaka) This stores unnecessary keyvalue pairs as well.
+			if isLaterEvent(apiListTimeStampMap, apiEvent.UUID+":"+env, currentTimeStamp) {
+				return
+			}
+			if strings.EqualFold(deployAPIToGateway, apiEvent.Event.Type) {
+				conf, _ := config.ReadConfigs()
+				for _, configuredEnv := range conf.ControlPlane.EventHub.EnvironmentLabels {
+					if configuredEnv == env {
+						if _, ok := subscription.APIListMap[env]; ok {
+							apiListOfEnv := subscription.APIListMap[env].List
+							for i := range apiListOfEnv {
+								// If API is already found, it is a new revision deployement.
+								// Subscription relates details of an API does not change between new revisions
+								if apiEvent.Context == apiListOfEnv[i].Context && apiEvent.Version == apiListOfEnv[i].Version {
+									logger.LoggerMsg.Debugf("APIList for apiIId: %s is not updated as it already exists", apiEvent.UUID)
+									return
+								}
+							}
+							queryParamMap := make(map[string]string, 3)
+							queryParamMap[subscription.GatewayLabelParam] = configuredEnv
+							queryParamMap[subscription.ContextParam] = apiEvent.Context
+							queryParamMap[subscription.VersionParam] = apiEvent.Version
+							// TODO: (VirajSalaka) Fix the REST API call once the APIM Event hub implementation is fixed.
+							go subscription.InvokeService(subscription.ApisEndpoint, subscription.APIListMap[env], queryParamMap,
```

Comment:
Use a single label and do the REST API call, since for each request, the same response is recieved.

Reply:
https://github.com/wso2/product-apim/issues/9993

If we make this change, it wont be necessary to call the service again at all. Hence the suggestion is not applied.